### PR TITLE
Resolve incorrect partial lengths state that prevented insert operation

### PR DESCRIPTION
**Pull request by jzaffiro**
This PR addresses the insert bug that appeared in the rollbackFarm tests. Insert ops that were created to be rolled back were being marked as removed without ever being acked. markRangeRemoved updates the partialLengths by subtracting the length of the node that was inserted, but since the length was never added to the tree in the first place, this resulted in negative node lengths. 

[AB#1809](https://dev.azure.com/fluidframework/internal/_workitems/edit/1809)

### CODE REVIEW

File: packages/dds/merge-tree/src/mergeTree.ts [hunk 1]
```diff
@@ -388,29 +388,29 @@ export interface IMergeTreeOptions {
     /**
      * Whether to enable the length calculations implemented in
      * https://github.com/microsoft/FluidFramework/pull/11678
-     * 
+     *
```
Comment (jzaffiro) [inline packages/dds/merge-tree/src/mergeTree.ts:391]: I tried to remove these spacing changes, but it removed the rest of my changes in this file as well. I then tried removing all of the changes and adding back just the ones I needed for this fix, but the spacing changes reappeared in the file. Not sure why they keep coming back, but I will continue trying to revert them.

Comment (alexvy86) [reply packages/dds/merge-tree/src/mergeTree.ts:391]: Auto-formatting removing trailing spaces is fine IMO.

**Comment by scarlettjlee** [comment]
Keep an eye out for when this merges to next. I get a conflict or test failure every time I touch mergeTree since the implementation has diverged. And I probably caused a merge conflict for you by modifying the minLength in rollback farm in next because it was hitting the bug you're fixing at length 16 (I don't understand why). If rollback farm is still failing with this change in next, just disable it, if possible, and open a new bug.

File: packages/dds/merge-tree/src/test/client.rollbackFarm.spec.ts [hunk 2]
```diff
@@ -40,7 +40,7 @@ describe("MergeTree.Client", () => {
                 mt.seedWithArray([0xDEADBEEF, 0xFEEDBED, minLength, opsPerRollback]);
 
                 // A: readonly, B: rollback, C: rollback + edit, D: edit
-                const clients = createClientsAtInitialState({ initialState: "" }, "A", "B", "C", "D");
+                const clients = createClientsAtInitialState({ initialState: "", options: {mergeTreeUseNewLengthCalculations: true} }, "A", "B", "C", "D");
```
Comment (jzaffiro) [inline packages/dds/merge-tree/src/test/client.rollbackFarm.spec.ts:43]: I originally forgot to take this out as this was one thing I tried as a fix, but would it be beneficial to leave it?

File: packages/dds/merge-tree/src/test/client.rollback.spec.ts [hunk 3]
```diff
@@ -51,6 +51,21 @@ describe("client.rollback", () => {
...
+
+        assert.equal(client.getText(), "defghi");
+        validatePartialLengths(client.getClientId(), client.mergeTree);
+
+        client.rollback?.({ type: MergeTreeDeltaType.INSERT }, client.peekPendingSegmentGroups());
+
+        assert.equal(client.getText(), "ghi");
+        validatePartialLengths(client.getClientId(), client.mergeTree);
+    });
     it("Should rollback multiple inserts with split segments", () => {
```
Comment (scarlettjlee) [inline packages/dds/merge-tree/src/test/client.rollback.spec.ts:69]: Would it be useful to validatePartialLengths in this test too?

File: packages/dds/merge-tree/src/test/client.rollbackFarm.spec.ts (hunk 2 above)
Comment (scarlettjlee) [inline packages/dds/merge-tree/src/test/client.rollbackFarm.spec.ts:43]: So anyone who uses rollback on mergeTree should set this option? If so, maybe we should set this in unit tests too.

File: packages/dds/merge-tree/src/test/client.rollback.spec.ts (hunk 3 above)
Comment (jzaffiro) [reply packages/dds/merge-tree/src/test/client.rollback.spec.ts:69]: It most likely wouldn't hurt, I know that validatePartialLengths can be an expensive operation but calling it once here should be fine.

File: packages/dds/merge-tree/src/test/client.rollbackFarm.spec.ts (hunk 2 above)
Comment (scarlettjlee) [reply packages/dds/merge-tree/src/test/client.rollbackFarm.spec.ts:43]: It looks like there will be an effect for this setting, but I'd err on the side of keeping the default to match what customers will likely have. So remove it if we don't absolutely need it.

**Review by scarlettjlee — APPROVED** [review]

**Pull request merged**